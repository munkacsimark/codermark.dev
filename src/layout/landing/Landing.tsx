import React from 'react';
import {
  viewNames,
  handleNavigation,
} from '../../view-router';
import Particles, { IParticlesParams } from 'react-particles-js';
import particlesConfig from './particles-config.json';
import styles from './Landing.module.scss';

const Landing: React.FunctionComponent = () => {
  return (
    <section
      id={viewNames.LANDING}
      className={styles.section}>
      <Particles
        className={styles.particles}
        width="100vw"
        height="100vh"
        params={particlesConfig as IParticlesParams}/>
      <h1 className={styles.greeting}>
          Hello, I'm <span className={styles.name}>Márk Munkácsi</span>.
          <br />
          I'm a frontend web developer.
      </h1>
      <a
        className={styles.showMore}
        href={`/#${viewNames.ABOUT}`}
        onClick={(event) => handleNavigation(event, viewNames.ABOUT)}>
        Show me more
        <svg
          className={styles.showMoreArrow}
          xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"/>
          <path fill="none" d="M0 0h24v24H0V0z"/>
        </svg>
      </a>
    </section>
  );
}

export default Landing;
