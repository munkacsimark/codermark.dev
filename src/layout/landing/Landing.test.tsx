import React from 'react';
import renderer from 'react-test-renderer';
import Landing from './Landing';

it('renders correctly', () => {
  const tree = renderer.create(<Landing />).toJSON();
  expect(tree).toMatchSnapshot();
});
