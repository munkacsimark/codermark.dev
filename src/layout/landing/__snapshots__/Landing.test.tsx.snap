// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="section"
  id="home"
>
  <div
    className="particles"
    id="tsparticles"
  >
    <canvas
      style={
        Object {
          "height": "100vh",
          "width": "100vw",
        }
      }
    />
  </div>
  <h1
    className="greeting"
  >
    Hello, I'm 
    <span
      className="name"
    >
      Márk Munkácsi
    </span>
    .
    <br />
    I'm a frontend web developer.
  </h1>
  <a
    className="showMore"
    href="/#about"
    onClick={[Function]}
  >
    Show me more
    <svg
      className="showMoreArrow"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
    </svg>
  </a>
</section>
`;
